### PR TITLE
Prow: Add build-e2e job for CAPM3

### PR DESCRIPTION
**Pull request by lentzi90**

### CODE REVIEW

**Review by tuminoid — APPROVED** [review]

File: prow/config/config.yaml [hunk 1]
```diff
@@ -641,6 +641,23 @@ presubmits:
...
+    - name: build-e2e
+      branches:
+        - main
+        - release-1.3
+      run_if_changed: '^test|^Makefile$'
+      decorate: true
+      spec:
+        containers:
+          - args:
+              - ./hack/unit.sh
```
Comment (mboukhalfa) [inline prow/config/config.yaml:653]:
```suggestion
              - ./hack/build-e2e.sh
```

Comment (lentzi90) [reply prow/config/config.yaml]: Nice catch, thanks!

**Review by Rozzii — APPROVED** [review]

**Comment by lentzi90** [comment]
/hold
Discussion offline if we should have a separate job for this or not. It could be included in some other job. Alternatively, we could do a more generic `build` job.

**Comment by lentzi90** [comment]
Updated to make it a generic build job. See related PR here: https://github.com/metal3-io/cluster-api-provider-metal3/pull/827

Only remaining question for me is: should we run this always or only when "relevant" files are touched? I put it now to run if any files under `test` or `api` or the `Makefile` is changed.

My reasoning behind this is that api and test folders both contains code that we don't necessarily build otherwise. (We build it for running tests, but not "on its own".) The rest of the code is always built anyway for producing the binary.

**Review by tuminoid — APPROVED** [review]

**Comment by lentzi90** [comment]
Removed `release-1.3` branch from job since it doesn't have the relevant script or issues. Thanks @furkatgofurov7 for noticing!

**Review by tuminoid — APPROVED** [review]
/lgtm

**Review by furkatgofurov7 — APPROVED** [review]

**Comment by furkatgofurov7** [comment]
putting hold back to avoid the situation when e2e tests might be failing in https://github.com/metal3-io/cluster-api-provider-metal3/pull/827 but this is merged, making any other PR fail for this job since script will land with 827. Will unhold when capm3 PR ready to take in. 
/hold

**Review by furkatgofurov7 — COMMENTED** [review]
Readding lgtm since it was removed during rebasing 

/lgtm

**Pull request merged**